### PR TITLE
Feat/remove folder

### LINKED ISSUE #129: Remove folder

### PR DESCRIPTION
**Pull request by kaitoz11**

### CODE REVIEW

File: src/Application/Folders/Commands/RemoveFolder.cs [hunk 1]
```diff
@@ -1,12 +1,61 @@
...
 public class RemoveFolder
 {
+    public class Validator : AbstractValidator<Command>
+    {
+        public Validator()
+        {
+            RuleLevelCascadeMode = CascadeMode.Stop;
+
+            RuleFor(f => f.FolderId)
+                .NotEmpty().WithMessage("FolderId is required.");
```
Comment (ChienNQuang) [inline src/Application/Folders/Commands/RemoveFolder.cs]: > This is not needed

File: src/Application/Folders/Commands/RemoveFolder.cs [hunk 2]
```diff
@@ -1,12 +1,61 @@
...
+            if (folder is null)
+            {
+                throw new KeyNotFoundException("Folder does not exist.");
+            }
+
+            var containDocument = folder.NumberOfDocuments > 0;
+
+            if (containDocument)
+            {
+                throw new InvalidOperationException("Folder cannot be removed because it contains documents.");
```
Comment (ChienNQuang) [inline src/Application/Folders/Commands/RemoveFolder.cs]: why still invalid operation exception

File: src/Application/Folders/Commands/RemoveFolder.cs [hunk 3]
```diff
@@ -1,12 +1,61 @@
...
+            }
+
```
Comment (ChienNQuang) [inline src/Application/Folders/Commands/RemoveFolder.cs:47]: you forgot to decrease number of folders in locker by 1

File: tests/Application.Tests.Integration/Folders/Commands/RemoveFolderTests.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        var command = new RemoveFolder.Command()
+        {
+            FolderId = folder.Id
+        };
+        
+        // Act
+        var action = async () => await SendAsync(command);
+        
+        // Assert
+        await action.Should().ThrowAsync<InvalidOperationException>()
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Folders/Commands/RemoveFolderTests.cs]: remember to change this exception

**Review by ChienNQuang — CHANGES_REQUESTED** [review]

File: src/Application/Folders/Commands/RemoveFolder.cs (hunk 2 above)
Comment (kaitoz11) [reply src/Application/Folders/Commands/RemoveFolder.cs]: fixed at https://github.com/ProFile-org/ProFile-BE/pull/141/commits/83a52ab2bcaf9c7b743b3e2d751068c72160f7c2

File: src/Application/Folders/Commands/RemoveFolder.cs (hunk 3 above)
Comment (kaitoz11) [reply src/Application/Folders/Commands/RemoveFolder.cs:47]: fixed at https://github.com/ProFile-org/ProFile-BE/pull/141/commits/83a52ab2bcaf9c7b743b3e2d751068c72160f7c2

File: tests/Application.Tests.Integration/Folders/Commands/RemoveFolderTests.cs (hunk 4 above)
Comment (kaitoz11) [reply tests/Application.Tests.Integration/Folders/Commands/RemoveFolderTests.cs]: fixed at https://github.com/ProFile-org/ProFile-BE/pull/141/commits/83a52ab2bcaf9c7b743b3e2d751068c72160f7c2

File: src/Application/Folders/Commands/RemoveFolder.cs (hunk 2 above)
Comment (kaitoz11) [reply src/Application/Folders/Commands/RemoveFolder.cs]: fixed at https://github.com/ProFile-org/ProFile-BE/pull/141/commits/83a52ab2bcaf9c7b743b3e2d751068c72160f7c2

File: src/Application/Folders/Commands/RemoveFolder.cs (hunk 1 above)
Comment (kaitoz11) [reply src/Application/Folders/Commands/RemoveFolder.cs]: fixed at https://github.com/ProFile-org/ProFile-BE/pull/141/commits/83a52ab2bcaf9c7b743b3e2d751068c72160f7c2

**Review by ChienNQuang — APPROVED** [review]
LGBT

**Pull request merged**